prow/test/integration/cmd/fakegcsserver/main.go

```diff
@@ -0,0 +1,131 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// fakegcsserver runs the open source GCS emulator from
+// https://github.com/fsouza/fake-gcs-server.
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+
+	"github.com/fsouza/fake-gcs-server/fakestorage"
+	"github.com/sirupsen/logrus"
+
+	configflagutil "k8s.io/test-infra/prow/flagutil/config"
+	"k8s.io/test-infra/prow/interrupts"
+	"k8s.io/test-infra/prow/logrusutil"
+	"k8s.io/test-infra/prow/pjutil"
+)
+
+type options struct {
+	// config is the Prow configuration. We need this to read in GCS
+	// configurations under plank's default_decoration_config_entries field set
+	// in the integration test's Prow configuration, because we have to
+	// initialize (create) these buckets before we upload into them with
+	// initupload.
+	config              configflagutil.ConfigOptions
+	emulatorPort        uint
+	emulatorPublicHost  string
+	emulatorStorageRoot string
+}
+
+func (o *options) validate() error {
+	if o.emulatorPort > 65535 {
+		return fmt.Errorf("-emulator-port range (got %d) must be 0 - 65535", o.emulatorPort)
+	}
+
+	return nil
+}
+
+func flagOptions() *options {
+	o := &options{config: configflagutil.ConfigOptions{ConfigPath: "/etc/config/config.yaml"}}
+	fs := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+
+	fs.UintVar(&o.emulatorPort, "emulator-port", 8888, "Port to use for the GCS emulator.")
+	fs.StringVar(&o.emulatorPublicHost, "emulator-public-host", "fakegcsserver.default:80", "Address of the GCS emulator as seen by other Prow components in the test cluster.")
+	fs.StringVar(&o.emulatorStorageRoot, "emulator-storage-root", "/gcs", "Folder to store GCS buckets and objects.")
+	o.config.AddFlags(fs)
+
+	fs.Parse(os.Args[1:])
+
+	return o
+}
+
+func main() {
+	logrusutil.ComponentInit()
+
+	o := flagOptions()
+	if err := o.validate(); err != nil {
+		logrus.WithError(err).Fatal("Invalid arguments.")
+	}
+
+	health := pjutil.NewHealth()
+	health.ServeReady()
+
+	defer interrupts.WaitForGracefulShutdown()
```

Comment:
This is copy/pasted code from other components that use our internal interrupts package. I guess we don't need this at all. Removed.

Reply:
@chaodaiG The graceful shutdown will wait up to 1 minute only if the other running goroutines need that time to perform additional cleanup work. As it stands, the goroutine spawned by fakestorage to start the server will always terminate immediately, so there won't be a delay.

I've tested this locally by running this program with `go run main.go -config-path config.yaml -emulator-storage-root $PWD/gcs` where both the config.yaml file and gcs folder is empty. As soon as you press CTRL-C to send a SIGTERM, `main` here will exit immediately. The same is true if you shut down the entire KIND cluster with `./teardown.sh -all` --- it will take about 15 seconds on my machine.

Reply:
> This is copy/pasted code from other components that use our internal interrupts package. I guess we don't need this at all. Removed.

So actually we do need this because otherwise `main` will exit immediately (tested and observed locally).